src/integration-test/java/com/mongodb/jdbc/integration/testharness/IntegrationTestUtils.java

```diff
@@ -0,0 +1,429 @@
+            }
+        }
+        if (test.expected_is_signed != null) {
+            assertEquals(test.expected_is_signed.size(), columnCount);
+            for (int i = 0; i < columnCount; i++) {
+                assertEquals(test.expected_is_signed.get(i), rsMetaData.isSigned(i + 1));
+            }
+        }
+        if (test.expected_is_writable != null) {
+            assertEquals(test.expected_is_writable.size(), columnCount);
+            for (int i = 0; i < columnCount; i++) {
+                assertEquals(test.expected_is_writable.get(i), rsMetaData.isWritable(i + 1));
+            }
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private static void validateResultsOrdered(TestEntry testEntry, ResultSet rs)
+            throws SQLException {
+        Integer actualRowCounter = null;
+        if (testEntry.expected_result != null) {
+            while (rs.next()) {
+                if (actualRowCounter == null) {
+                    actualRowCounter = 0;
+                }
+                assertTrue(
+                        compareRow(
+                                (List<Object>) testEntry.expected_result.get(actualRowCounter),
+                                rs));
+                actualRowCounter++;
+            }
+        }
+        if (testEntry.row_count != null) {
+            validateRowCount(testEntry.row_count, testEntry, actualRowCounter, rs);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private static void validateResultsUnordered(TestEntry testEntry, ResultSet rs)
+            throws SQLException {
+        Integer actualRowCounter = null;
+        if (testEntry.expected_result != null) {
+            while (rs.next()) {
+                boolean found = false;
+                if (actualRowCounter == null) {
+                    actualRowCounter = 0;
+                }
+                actualRowCounter++;
+                for (Object expectedRow : testEntry.expected_result) {
+                    if (compareRow((List<Object>) expectedRow, rs)) {
+                        found = true;
+                        break;
+                    }
+                }
+                assertTrue(found);
+            }
+        }
+        if (testEntry.row_count != null) {
+            validateRowCount(testEntry.row_count, testEntry, actualRowCounter, rs);
+        }
+    }
+
+    private static void validateRowCount(
+            int expectedRowCount, TestEntry testEntry, Integer actualRowCounter, ResultSet rs)
+            throws SQLException {
+        if (actualRowCounter == null) {
+            actualRowCounter = MongoIntegrationTest.countRows(rs);
+        }
+        if (testEntry.row_count_gte != null && testEntry.row_count_gte) {
+            assertTrue(actualRowCounter >= expectedRowCount);
+        } else {
+            assertEquals(java.util.Optional.ofNullable(actualRowCounter), expectedRowCount);
```

Comment:
```suggestion
    private static void validateRowCount(
            TestEntry testEntry, Integer actualRowCounter, ResultSet rs)
            throws SQLException {
        if (actualRowCounter == null) {
            actualRowCounter = MongoIntegrationTest.countRows(rs);
        }
        if (testEntry.row_count_gte != null && testEntry.row_count_gte) {
            assertTrue(actualRowCounter >= testEntry.row_count);
        } else {
            assertEquals(java.util.Optional.ofNullable(actualRowCounter), testEntry.row_count);
```
[nit] I don't think we should treat `testEntry.row_count` special by making it separate argument from `testEntry`, especially if we don't do the same for `testEntry.row_count_gte`. What I mean is, the signature should either be:
```
private static void validateRowCount(int expectedRowCount, Boolean expectedRowCountGte, Integer actualRowCount, ResultSet rs)
```
or
```
private static void validateRowCount(TestEntry testEntry, Integer actualRowCount, ResultSet rs)
```

I don't have a preference between the two, but think it is odd to pass `(testEntry.row_count, testEntry, ...)`.

Reply:
Ok sounds good.  I implemented the latter: 
```
private static void validateRowCount(TestEntry testEntry, Integer actualRowCount, ResultSet rs)
```